.github/workflows/ci.yml

```diff
@@ -0,0 +1,85 @@
+---
+name: OpenWISP Users CI Build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    name: Python==${{ matrix.python-version }} | ${{ matrix.django-version }}
+    runs-on: ubuntu-20.04
+
+    services:
+      redis:
+        image: redis
+        ports:
+          - 6379:6379
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - 3.6
+          - 3.7
+        django-version:
+          - django~=2.2
+          - django~=3.0
+          - django~=3.1
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install python dependencies
+        run: |
+          pip install -U pip wheel
+          pip install ${{ matrix.django-version }}
+          pip install -U -r requirements-test.txt
+      - name: Install npm dependencies
+        run: sudo npm install -g jshint
+
+      - name: Install openwisp-users
+        run: pip install -e .[rest]
+
+      - name: QA checks
+        run: |
+          ./run-qa-checks
+
+      - name: Tests
+        run: |
+          coverage run --source=openwisp_users runtests.py
+          SAMPLE_APP=1 coverage run --append --source=openwisp_users runtests.py
+          # basic tests without allauth.socialaccount
+          NO_SOCIAL_APP=1 ./tests/manage.py test testapp.tests.test_admin.TestUsersAdmin --parallel
+
+      - name: Upload Coverage
+        run: coveralls --service=github
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          COVERALLS_FLAG_NAME: |
+            python-${{ matrix.python-version }}-${{ matrix.django-version }}
+          COVERALLS_PARALLEL: true
+
+  coveralls:
+    name: Finish Coveralls
+    needs: build
+    runs-on: ubuntu-latest
+    container: python:3-slim
+    steps:
+      - name: Finished
+        run: |
+          pip3 install --upgrade coveralls
```

Comment:
@Purhan I think this line is not needed as coveralls will be updated while installing from `requirements-test.txt`

Reply:
@devkapilbansal are you sure about this? If I know correctly, all jobs supposedly run on different containers as soon as machines are available to execute them.

You can look at [this build](https://github.com/openwisp/openwisp-controller/runs/1696220231) where it actually installs coveralls, otherwise would've prompted that it's already installed. Correct me if I'm wrong.

Reply:
@Purhan he's right, coveralls is defined as a dependency of `openwisp-utils[qa]`: https://github.com/openwisp/openwisp-utils/blob/master/setup.py#L51

Reply:
@Purhan he's right, coveralls is defined as a dependency of `openwisp-utils[qa]`: https://github.com/openwisp/openwisp-utils/blob/master/setup.py#L51

Reply:
@Purhan is right here. All jobs run on different containers. Installing coveralls is needed.

@nemesisdesign @devkapilbansal look at workflows in other repositories of OpenWISP.

- [openwisp-controller](https://github.com/openwisp/openwisp-controller/blob/master/.github/workflows/ci.yml#L81-L92)
- [openwisp-notifications](https://github.com/openwisp/openwisp-notifications/blob/master/.github/workflows/build.yml#L79-L90)
- [openwisp-radius](https://github.com/openwisp/openwisp-radius/blob/master/.github/workflows/ci.yml#L68-L79)
- [openwisp-monitoring](https://github.com/openwisp/openwisp-monitoring/blob/master/.github/workflows/ci.yml#L110-L121)

Reply:
oh, I didn't know that, thanks for explaining

Reply:
Thanks for correcting me 😅